### PR TITLE
[wasm] Copy source files into the runtime pack, they will be needed by AOT in the future.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/wasm/Makefile [hunk 1]
```diff
@@ -91,6 +92,28 @@ build-interp-$(1): $(BUILDS_BIN_DIR)/$(1)/dotnet.js
...
 endef
 
+# Needed by AOT
+
+SRC_FILES=\
+	runtime/driver.c \
+	runtime/corebindings.c \
+	runtime/binding_support.js \
+	runtime/dotnet_support.js \
+	runtime/library_mono.js
```
Comment (akoeplinger) [inline src/mono/wasm/Makefile:102]: these .js files will need to be added to https://github.com/dotnet/runtime/blob/master/eng/SignCheckExclusionsFile.txt#L15 to workaround the signing issue.

File: src/mono/wasm/Makefile [hunk 2]
```diff
@@ -91,6 +92,28 @@ build-interp-$(1): $(BUILDS_BIN_DIR)/$(1)/dotnet.js
...
+
+DST_FILES=$(patsubst runtime/%,$(BUILDS_SRC_DIR)/%, $(SRC_FILES))
```
Comment (marek-safar) [inline src/mono/wasm/Makefile:104]: I don't think this is the best approach, we'll have most likely different runtime pack for aot configuration(s).

Comment (vargaz) [reply src/mono/wasm/Makefile:104]: What will be in those packs ?

Comment (marek-safar) [reply src/mono/wasm/Makefile:104]: For example different code because SRE won't be supported, AOT specific binding, wasm version of VM, etc

**Pull request closed without merging**